code/modules/holomap/holomap.dm

```diff
@@ -0,0 +1,233 @@
+/*
+	Somone should totally add an action button to select a location when observing the map
+	and it display a list to select from (TGUI maybe!? color coded !? YES?)
+	then show an icon like YouAreHere there
+*/
+
+/obj/machinery/holomap
+	name = "Holomap"
+	desc = "It displays a holographic map of the station."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "map"
+	anchored = TRUE
+	idle_power_consumption = 5
+	active_power_consumption = 15
+	var/display_map //for mappers who want to add holomaps for off station locations, otherwise it gets set in Initialize()
+	var/list/display_targets = list()
+	var/image/holomap_projection
+	var/image/YouAreHere
+
+/obj/machinery/holomap/Initialize(mapload)
+	. = ..()
+	if(!display_map)
+		display_map = SSmapping.map_datum.technical_name
+	var/icon/base_icon = make_map()
+	var/icon/alpha_mask = icon('icons/effects/480x480.dmi', "scanline") //Scanline effect.
+	base_icon.AddAlphaMask(alpha_mask)
+	holomap_projection = image(base_icon, loc = src, layer = ABOVE_HUD_PLANE, pixel_x = -240, pixel_y = -240)
+	holomap_projection.plane = ABOVE_HUD_PLANE
+	holomap_projection.filters = filter(type = "drop_shadow", size = 8, color = "#ffffff65")
+	YouAreHere = image('icons/effects/holomap_icons.dmi', src, "YouAreHere", 22.1 , 2, -256+(x*2), -256+(y*2))
+	YouAreHere.plane = 22.1
```

Comment:
I imagine this is all going to be fairly heavy in initialize performance wise. I think we should move this to its own proc, and call it 5 seconds or so after initalizing, so it can be created while the server is in the OOC lobby (or something like that, I'm just trying to think of how IDs do it to minimize time lost during roundstart/init)

Reply:
Wouldn't running it with everything else in the giant red wall of text be ideal? Why no?

Reply:
Just to minimize init times, if we can do stuff with it when we're doing nothing else (just the server waiting and OOC chatting in lobby), we have plenty more computational resources to use while we wait. Probably more than 5 seconds given how long init is... Don't feel obligated to implement this if you're not up to it. Just an idea to improve performance